site/docs/skylark/windows_tips.md

```diff
@@ -0,0 +1,264 @@
+---
+layout: documentation
+title: Writing rules on Windows
+---
+
+# Writing rules on Windows
+
+Common problems of writing platform-independent rules and ideas to overcome them.
+
+This document focuses on writing rules that work on Windows.
+
+## Paths
+
+Problems:
+
+- **Length limit**: maximum path length is 259 characters.
+
+  While Windows also supports paths up to 32767 characters ("long paths"), many programs are built
+  with the lower limits. The action your rule executes may not handle long paths either.
+
+- **Working directory**: is also limited to 259 characters.
+
+  Processes cannot `cd` into a directory longer than 259 characters.
+
+- **Case-sensitivity**: Windows paths are case-insensitive, Unix paths are case-sensitive.
+
+  Be aware of this when creating command lines for actions.
+
+- **Path separators**: are backslash (`\`), not forward slash (`/`).
+
+  Bazel uses paths with `/` separators, so on Windows you have to replace those with `\` when
+  creating comamnd lines and environment variables for actions.
+
+- **Absolute paths**: don't start with slash (`/`).
+
+  Absolute paths on Windows start with a drive letter, e.g. `C:\foo\bar.txt`. There's no single
+  filesystem root.
+
+  Be aware of this if your rule checks if a path is absolute.
```

Comment:
In general, absolute paths should be avoided though. Absolute paths are often non-portable.